tests-hivemq-platform-operator/src/integrationTest/java/com/hivemq/helmcharts/compose/HelmLegacyStatefulSetMigrationIT.java

```diff
@@ -0,0 +1,171 @@
+        final var legacyOperatorDeploymentName = String.format("%s-hivemq-operator-operator", LEGACY_RELEASE_NAME);
+        K8sUtil.scaleDeployment(client, operatorNamespace, legacyOperatorDeploymentName, 0);
+        var legacyOperatorLabels = K8sUtil.getHiveMQLegacyOperatorLabels(LEGACY_RELEASE_NAME);
+        client.pods()
+                .inNamespace(operatorNamespace)
+                .withLabels(legacyOperatorLabels)
+                .waitUntilCondition(Objects::isNull, 1, TimeUnit.MINUTES);
+        assertThat(client.pods()
+                .inNamespace(operatorNamespace)
+                .withLabels(legacyOperatorLabels)
+                .list()
+                .getItems()).isEmpty();
+
+        // delete custom resource with orphan
+        // kubectl delete hivemq-clusters.hivemq.com <release-name> --cascade=orphan -n <namespace>
+        final var legacyPlatform = K8sUtil.getLegacyHiveMQPlatform(client, operatorNamespace, LEGACY_RELEASE_NAME);
+        legacyPlatform.withPropagationPolicy(DeletionPropagation.ORPHAN).withTimeoutInMillis(5000).delete();
+        assertThat(K8sUtil.getLegacyHiveMQPlatform(client, operatorNamespace, LEGACY_RELEASE_NAME).get()).isNull();
+
+        // scale up the legacy operator
+        // kubectl scale deployment <release-name>-hivemq-operator-operator --replicas=1 -n <namespace>
+        K8sUtil.scaleDeployment(client, operatorNamespace, legacyOperatorDeploymentName, 1);
+        K8sUtil.waitForLegacyOperatorPodStateRunning(client, operatorNamespace, LEGACY_RELEASE_NAME);
+        assertThat(client.pods()
+                .inNamespace(operatorNamespace)
+                .withLabels(legacyOperatorLabels)
+                .list()
+                .getItems()
+                .size()).isEqualTo(1);
+
+        //TODO: We cannot execute `helm uninstall my-release --cascade orphan` - See: https://github.com/helm/helm/issues/13279
+        // we need to uninstall the legacy release with orphan option so the new platform Helm install takes over existing resources
+        //helmChartContainer.uninstallRelease(LEGACY_RELEASE_NAME, operatorNamespace, "--cascade", "orphan");
+
+        // as a workaround we need to delete the Helm secret installed
+        // kubectl delete secret -l owner=helm,name=<release-name> -n <namespace>
+        final var helmSecretLabels = Map.of("owner", "helm", "name", LEGACY_RELEASE_NAME);
+        client.secrets().inNamespace(operatorNamespace).withLabels(helmSecretLabels).withTimeoutInMillis(5000).delete();
+        assertThat(client.secrets()
+                .inNamespace(operatorNamespace)
+                .withLabels(helmSecretLabels)
+                .list()
+                .getItems()).isEmpty();
+
+        // create the corresponding config.xml file in the new ConfigMap for the platform operator
+        K8sUtil.createConfigMap(client, operatorNamespace, "hivemq-migration-config-map.yml");
```

Comment:
Hmm, I have to do this in the operator IT because there is no Helm chart.
But shouldn't the goal of this IT be that a matching broker `config.xml` is automatically created by the Helm chart?

Wouldn't that also be the way customers would approach the migration? They need to transfer their custom settings from the legacy to the platform `values.yaml`. And it's our your job to offer all config options they might want to transfer.

IMO we should not rely on a `config.xml` being provided.

Reply:
Is this even used? I cannot see a `--set-file config.overrideHiveMQConfig=hivemq-migration-config-map.xml` anywhere.

Reply:
Ahh, it is used because we explicitly don't create a new config:
```yaml
config:
  create: false
  name: "hivemq-configuration"
```

Is there a config option missing to let the Helm chart create the config?

Reply:
That's a very valid point, and I believe it's going to be the most painful issue for customers regarding migration.

My understanding here was that it would be the customers who will responsible for providing the right `config.xml` file (based on their own legacy values used, environment variable used for the placeholders, etc.). So I tried to follow the same approach, and at the same time I didn't want to go too deep into using a custom `config.xml` file for that, so I'm using just only the default values.

But you're right, the IT should aim to mimic how customer should try to match all the options from the legacy chart to the new one regarding the `config.xml` file generation and then handle it from there. Specially important when it comes to the listeners defined in a custom `config.xml` as these need to match with the ones defined in values YAML file (`services` section).

> Is there a config option missing to let the Helm chart create the config?

I don't think so. As I mentioned, I tried to use what I thought it would be the most typical use case, but it makes more sense to generate a configuration file based on the values set in a customer values YAML file rather than relying on a provided custom configuration file.

Will change that and create a custom `config.xml` from legacy and its counterpart in the Platform chart.